extensions-contrib/opentelemetry-emitter/src/main/java/org/apache/druid/emitter/opentelemetry/DruidContextTextMapGetter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.opentelemetry;
+
+import io.opentelemetry.context.propagation.TextMapGetter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Implementation of a text-based approach to read the W3C Trace Context from request.
+ * <a href="https://opentelemetry.io/docs/java/manual_instrumentation/#context-propagation">Context propagation</a>
+ * <a href="https://www.w3.org/TR/trace-context/">W3C Trace Context</a>
+ */
+public class DruidContextTextMapGetter implements TextMapGetter<ServiceMetricEvent>
```

Comment:
Adding the `extractedKeys` field means that this class is no longer threadsafe.  So we should add some annotations and/or javadoc comments to that effect.
```suggestion
@NotThreadSafe
public class DruidContextTextMapGetter implements TextMapGetter<ServiceMetricEvent>
```